### PR TITLE
New inputs

### PR DESCRIPTION
**Pull request by burakukula**
Changes:

 - All form elements have 2 variants, default is always gray, additionally, we have white version.
 - Sizes are set in two variants - normal and large. 
 - updates in search - 2 versions with gray icon and with black round button
 - small README for form elements - it supposes to be extended in the future
 
form-elements mov

### CODE REVIEW

File: src/components/form-elements/_textarea.scss [hunk 1]
```diff
@@ -77,23 +74,32 @@ $includeHtml: false !default;
...
+      height: 168px;
     }
 
     &--auto-height {
-      line-height: rhythm(1);
+      line-height: 24px;
       height: auto;
     }
+
+    &--nested {
```
Comment (matzimowski) [inline src/components/form-elements/_textarea.scss]: remove me

File: src/components/form-elements/pages/input-interactive.jsx [hunk 2]
```diff
@@ -48,17 +40,14 @@ const Inputs = () => {
...
     <div>
-      <DocsActiveBlock settings={settings}>
-        <Input placeholder="Simple input" />
-      </DocsActiveBlock>
-      <DocsActiveBlock settings={settings}>
+      <DocsActiveBlock backgroundColor="dark" settings={settings}>
         <Input
-          placeholder="Big wrong input"
+          backgroundColor="none"
+          placeholder="Big input"
```
Comment (matzimowski) [inline src/components/form-elements/pages/input-interactive.jsx]: maybe use a `large` word here too? wdyt? instead of `big`

Comment (burakukula) [reply src/components/form-elements/pages/input-interactive.jsx]: sure thing.

File: src/components/form-elements/_textarea.scss (hunk 1 above)
Comment (burakukula) [reply src/components/form-elements/_textarea.scss]: done

File: src/components/form-elements/pages/input-interactive.jsx (hunk 2 above)
Comment (burakukula) [reply src/components/form-elements/pages/input-interactive.jsx]: solved

File: src/components/form-elements/Input.jsx [hunk 3]
```diff
@@ -3,9 +3,9 @@
 import React from 'react';
 import classnames from 'classnames';
 
-type InputSizeType = 'small' | 'normal' | 'large';
+type InputSizeType = 'normal' | 'large';
 
-type InputColorType = 'normal' | 'light' | 'light-alt';
+type InputColorType = 'default' | 'white';
```
Comment (vergilius) [inline src/components/form-elements/Input.jsx]: so in one place we have `normal` and in other `default`, why is that?

Comment (burakukula) [reply src/components/form-elements/Input.jsx]: nope, colors are set to `default` and `white` and sizes to `normal` and `large`

File: src/components/form-elements/Input.jsx [hunk 4]
```diff
@@ -53,28 +53,67 @@ export const TYPE = Object.freeze({
...
   size?: ?InputSizeType,
+  /**
+   * There are two color variants for form elements, default does not have to be specified
+   * @example <Input color="white" placeholder="placeholder" />
+   * @see color="default" https://styleguide.brainly.com/latest/docs/interactive.html?color="default"#input
+   * @see color="white" https://styleguide.brainly.com/latest/docs/interactive.html?color="white"#input
+   */
   color?: ?InputColorType,
+  /**
+   * Optional additional boolean to specified if it's valid
```
Comment (vergilius) [inline src/components/form-elements/Input.jsx]: do we need both `Optional additional` next to each other?

Comment (burakukula) [reply src/components/form-elements/Input.jsx]: nope, we don't I will fix it. thx

File: src/components/form-elements/Input.jsx [hunk 5]
```diff
@@ -53,28 +53,67 @@ export const TYPE = Object.freeze({
...
+   * @example <Input valid placeholder="placeholder" />
+   */
   valid?: boolean,
+  /**
```
Comment (vergilius) [inline src/components/form-elements/Input.jsx]: would that be valid to have `valid` and `invalid` exclude each other? Right now it's not, and the CSS makes difference

File: src/components/form-elements/Input.jsx (hunk 4 above)
Comment (burakukula) [reply src/components/form-elements/Input.jsx]: done.

File: src/components/form-elements/Textarea.jsx [hunk 6]
```diff
@@ -5,25 +5,81 @@ import classnames from 'classnames';
...
+
 export const SIZE = {
   SHORT: 'short',
   NORMAL: 'normal',
   TALL: 'tall',
   XTALL: 'xtall',
 };
 
+export const TEXTAREA_COLOR = {
+  DEFAULT: 'default',
```
Comment (vergilius) [inline src/components/form-elements/Textarea.jsx]: isn't the no-option indicate the default behaviour (and look)?

File: src/components/form-elements/_select.scss [hunk 7]
```diff
@@ -1,40 +1,43 @@
-$selectHeight: rhythm(3 / 2);
-$selectBackground: $white;
+$selectHeight: 40px;
```
Comment (vergilius) [inline src/components/form-elements/_select.scss]: I'd like to see where these sizes come from

File: src/components/form-elements/Input.jsx (hunk 5 above)
Comment (burakukula) [reply src/components/form-elements/Input.jsx]: It never was. We can refactor the logic of the components, but maybe let's do it separately. WDYT? Your point is valid for more than just inputs.

File: src/components/form-elements/Textarea.jsx (hunk 6 above)
Comment (burakukula) [reply src/components/form-elements/Textarea.jsx]: We need it for export to sketch.

File: src/components/form-elements/_select.scss [hunk 8]
```diff
@@ -1,40 +1,43 @@
...
     font-size: $selectFontSize;
     color: $selectTextColor;
-    box-shadow: $componentShadow;
     border-radius: $selectBorderRadius;
     height: $selectHeight;
+    border: 2px solid $graySecondaryLightest;
 
     &__element {
+      position: relative;
+      top: -2px;
```
Comment (vergilius) [inline src/components/form-elements/_select.scss]: how about extract that to some `$borderRelatedVariable`? that would be clear what's happening here without need to check the border

File: src/components/form-elements/_select.scss (hunk 7 above)
Comment (burakukula) [reply src/components/form-elements/_select.scss]: this is the same size, as we have in Buttons, I can move it to config if you like.

File: src/components/form-elements/_select.scss (hunk 8 above)
Comment (burakukula) [reply src/components/form-elements/_select.scss]: 👍

File: src/components/form-elements/_select.scss [hunk 9]
```diff
@@ -43,35 +46,136 @@ $includeHtml: false !default;
...
+        height: $selectLargeHeight;
+        padding: 0 50px 0 spacing(m);
+        
+        &::-ms-expand {
+          display: none;
+        }
+      }
+
+      &::placeholder {
+        font-size: 18px;
```
Comment (vergilius) [inline src/components/form-elements/_select.scss]: why don't we reuse variables? like this `$inputLargeFontSize` - its 18px

File: src/components/form-elements/_text-input.scss [hunk 10]
```diff
@@ -1,148 +1,137 @@
...
+    &--white {
+      background: $white;
+      border: 2px solid $white;
+      border-radius: $inputStandardBorderRadius;
 
-      &::placeholder {
-        color: $white;
-        font-weight: $fontWeightBold;
+      &:hover {
+        background-color: mix($white, $graySecondaryUltraLight, 12%);
```
Comment (vergilius) [inline src/components/form-elements/_text-input.scss]: where does `12%` comes from? is the final color part of our palette?

File: src/components/form-elements/Input.jsx (hunk 3 above)
Comment (vergilius) [reply src/components/form-elements/Input.jsx]: is that a updated standard for colors? there was `normal` previously in component

File: src/components/form-elements/Input.jsx (hunk 5 above)
Comment (vergilius) [reply src/components/form-elements/Input.jsx]: gotit! 👍

File: src/components/form-elements/Textarea.jsx (hunk 6 above)
Comment (vergilius) [reply src/components/form-elements/Textarea.jsx]: oh I see

File: src/components/form-elements/_select.scss (hunk 7 above)
Comment (vergilius) [reply src/components/form-elements/_select.scss]: I mean, even comment would be enough, that base size of our components is 40px and that comes from designs for example

👍

File: src/components/form-elements/_text-input.scss [hunk 11]
```diff
@@ -1,148 +1,137 @@
...
+      &:active,
+      &:focus:hover,
+      &:active:hover {
+        background-color: $white;
+        border: 2px solid $inputFocusInvalidBorderColor;
+        box-shadow: none;
+      }
+
+      &:hover {
+        background-color: mix($inputBackground, $graySecondaryLight, 80%);
```
Comment (vergilius) [inline src/components/form-elements/_text-input.scss]: mix again, do we reuse these colors in other components?

File: src/components/form-elements/_text-input.scss (hunk 10 above)
Comment (burakukula) [reply src/components/form-elements/_text-input.scss]: it's not, this is the same pattern as we use for `Button`. For filled element, we use 80% mix and for light version 12%. I will try to describe it in some part of the README.

File: src/components/form-elements/_select.scss (hunk 9 above)
Comment (burakukula) [reply src/components/form-elements/_select.scss]: I am moving it into configuration. I will let you know as soon as it's refactored

File: src/components/form-elements/Input.jsx (hunk 3 above)
Comment (burakukula) [reply src/components/form-elements/Input.jsx]: yeah, this would be the new approach for form elements.

**Pull request merged**